stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/shared/media/ChatInfoSharedMediaFragment.kt

```diff
@@ -0,0 +1,202 @@
+package io.getstream.chat.ui.sample.feature.chat.info.shared.media
+
+import android.graphics.Rect
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.getstream.sdk.chat.utils.Utils
+import com.getstream.sdk.chat.view.EndlessScrollListener
+import io.getstream.chat.ui.sample.R
+import io.getstream.chat.ui.sample.common.initToolbar
+import io.getstream.chat.ui.sample.databinding.FragmentChatInfoSharedMediaBinding
+import io.getstream.chat.ui.sample.feature.chat.info.shared.ChatInfoSharedAttachmentsViewModel
+import io.getstream.chat.ui.sample.feature.chat.info.shared.ChatInfoSharedAttachmentsViewModelFactory
+import io.getstream.chat.ui.sample.feature.chat.info.shared.SharedAttachment
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class ChatInfoSharedMediaFragment : Fragment() {
+
+    private val args: ChatInfoSharedMediaFragmentArgs by navArgs()
+    private val factory: ChatInfoSharedAttachmentsViewModelFactory by lazy {
+        ChatInfoSharedAttachmentsViewModelFactory(
+            args.cid,
+            ChatInfoSharedAttachmentsViewModel.AttachmentsType.MEDIA
+        )
+    }
+    private val viewModel: ChatInfoSharedAttachmentsViewModel by viewModels { factory }
+    private val sharedMediaViewModel: ChatInfoSharedMediaViewModel by activityViewModels()
+    private val adapter: ChatInfoSharedMediaAdapter = ChatInfoSharedMediaAdapter()
+    private val scrollListener = EndlessScrollListener(LOAD_MORE_THRESHOLD) {
+        viewModel.onAction(ChatInfoSharedAttachmentsViewModel.Action.LoadMoreRequested)
+    }
+    private val dateScrollListener by lazy {
+        MediaDateScrollListener(
+            spanCount = SPAN_COUNT,
+            positionChangeThreshold = { binding.mediaRecyclerView.top - binding.dateTextView.bottom }
```

Comment:
It' because `dateScrollListener` is created in `onViewCreated` in which `binding.mediaRecyclerView.top` and `binding.dateTextView.bottom` are still equal to 0.